pytest_fixtures/component/discovery.py

```diff
@@ -0,0 +1,47 @@
+from copy import copy
+
+import pytest
+
+
+@pytest.fixture(scope='module')
+def module_discovery_sat(
+    module_provisioning_rhel_content, module_target_sat, module_sca_manifest_org, module_location
+):
+    """Creates a Satellite with discovery installed and configured"""
+    sat = module_target_sat
+    # Register to CDN and install discovery image
+    sat.register_to_cdn()
+    sat.execute('yum -y --disableplugin=foreman-protector install foreman-discovery-image')
+    sat.unregister()
+    # Symlink image so it can be uploaded for KEXEC
+    disc_img_path = sat.execute(
+        'find /usr/share/foreman-discovery-image -name "foreman-discovery-image-*.iso"'
+    )
+    disc_img_name = disc_img_path.stdout.split("/")[-1]
+    sat.execute(f'ln -s {disc_img_path} /var/www/html/pub/{disc_img_name}')
+    # Change 'Default PXE global template entry'
+    pxe_entry = sat.api.Setting().search(query={'search': 'Default PXE global template entry'})[0]
```

Comment:
You can use settings fixture that sets and unsets value for session/function(probably) scope.

Reply:
Wouldn't that require me to parametrize every test with the settings I'm going to change?